ApiBundle/Controller/ContentController.php

```diff
@@ -32,17 +34,44 @@ class ContentController extends BaseController
      * @Api\Serialize()
      *
      * @return FacadeInterface
+     * @throws ContentNotFoundHttpException
      */
     public function showAction(Request $request, $contentId)
     {
+        $content = $this->findOneContent($contentId, $request->get('language'), $request->get('version'));
+
+        if (!$content) {
+            throw new ContentNotFoundHttpException();
+        }
+
+        return $this->get('open_orchestra_api.transformer_manager')->get('content')->transform($content);
+    }
+
+    /**
+     * @param Request $request
+     * @param string  $contentId
+     *
+     * @Config\Route("/{contentId}/show-or-create", name="open_orchestra_api_content_show_or_create")
+     * @Config\Method({"GET"})
+     *
+     * @Config\Security("has_role('ROLE_ACCESS_CONTENT_TYPE_FOR_CONTENT')")
+     *
+     * @Api\Serialize()
+     *
+     * @return FacadeInterface
+     * @throws ContentSourceNotFoundHttpException
+     */
+    public function showOrCreateAction(Request $request, $contentId)
+    {
         $language = $request->get('language');
-        $version = $request->get('version');
-        $contentRepository = $this->get('open_orchestra_model.repository.content');
-        $content = $contentRepository->findOneByContentIdAndLanguageAndVersion($contentId, $language, $version);
+        $content = $this->findOneContent($contentId, $language, $request->get('version'));
 
         if (!$content) {
-            $defaultCurrentSiteLanguage = $this->get('open_orchestra_backoffice.context_manager')->getCurrentSiteDefaultLanguage();
-            $oldContent = $contentRepository->findOneByContentIdAndLanguageAndVersion($contentId, $defaultCurrentSiteLanguage);
+            $sourceLanguage = $request->get('source_language');
+            if (!$sourceLanguage) {
+                throw new ContentSourceNotFoundHttpException();
```

Comment:
I think the exception is still not correct : we can not say that the content is not found as we haven't search for it